File: scio-elasticsearch/src/main/java/com/spotify/scio/elasticsearch/ElasticsearchOptions.java
package com.spotify.scio.elasticsearch;

import java.net.InetSocketAddress;

public class ElasticsearchOptions {
  private final String clusterName;
  private final InetSocketAddress[] servers;
  public ElasticsearchOptions(String clusterName, InetSocketAddress[] servers) {
    this.clusterName = clusterName;
    this.servers = servers;
  }
  public String clusterName() {
    return clusterName;
  }
  public InetSocketAddress[] servers() {
    return servers;
  }
}

File: scio-elasticsearch/common/src/main/scala/com/spotify/scio/elasticsearch/ElasticsearchOptions.scala
package com.spotify.scio.elasticsearch

import co.elastic.clients.json.{JsonpMapper, SimpleJsonpMapper}
import org.apache.http.HttpHost

final case class ElasticsearchOptions(
  nodes: Seq[HttpHost],
  usernameAndPassword: Option[(String, String)] = None,
  mapperFactory: () => JsonpMapper = () => new SimpleJsonpMapper()
)
